_rules/SC2-4-4-link-has-name.md

```diff
@@ -0,0 +1,157 @@
+---
+name: Links have an accessible name
+description: |
+  Each link has an accessible name
+
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects: # Remove what is not applicable
+- DOM Tree
+- CSS Styling
+
+authors:
+- Wilco Fiers
+- Anne Thyme Nørregaard
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to any HTML or SVG element with the [semantic role](#semantic-role) of `link` that is [visible on the page](#visible-on-the-page) or [exposed to assistive technologies](#exposed-to-assistive-technologies).
+
+### Expectation
+
+Each target element has an [accessible name](#accessible-name) that is [non-empty](#non-empty).
+
+## Assumptions
+
+- The rule assumes that all links are user interface components as defined by WCAG 2. When the link role is used incorrectly, this assumption may not be true.
+
+## Accessibility Support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- [https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=244#navigation-mechanisms-refs](https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=244#navigation-mechanisms-refs)
+- [ARIA7: Using aria-labelledby for link purpose](https://www.w3.org/TR/WCAG20-TECHS/ARIA7.html)
+- [ARIA8: Using aria-label for link purpose](https://www.w3.org/TR/WCAG20-TECHS/ARIA8.html)
+- [F89: Failure of Success Criteria 2.4.4, 2.4.9 and 4.1.2 due to using null alt on an image where the image is the only content in a link](http://www.w3.org/TR/WCAG20-TECHS/F89.html)
+
+## Test Cases
+
+### Passed
+```html
+<!-- <a> element with accessible name through content -->
+<a href="http://www.w3.org/WAI"> Web Accessibility Initiative (WAI) </a>
+```
+
+```html
+<!-- element with eplicit role of link with accessible name through content -->
+<div role="link"> Web Accessibility Initiative (WAI) </div>
+```
+
+```html
+<!-- Button with the role of link -->
+<button role="link">Click me!</button>
+```
+
+```html
+<!-- Accessible name for link via aria-label -->
+<a href="http://www.w3.org/WAI"><img src="#" aria-label="This is a link"/></a>
+```
+
+```html
+<!-- Link named via title on link -->
+<a href="http://www.w3.org/WAI" title="This is a link"><img src="#" /></a>
+```
+
+```html
+<!-- Link named via title on image -->
+<a href="http://www.w3.org/WAI"><img src="#" title="This is a link"/></a>
+```
+
+```html
+<!-- Link with both image and text -->
+<a href="http://www.w3.org/WAI"><img src="#" />This is my link text</a>
+```
+
+```html
+<!-- Accessible name for link via aria-labelledby -->
+<a href="http://www.w3.org/WAI"><img src="#" aria-labelledby="id1"/></a>
+<div id="id1">This is my link text</div>
+```
+
+```html
+<!-- Passing because this rule doesn't consider whether the accessible name is descriptive enough -->
```

Comment:
I don't think this is a useful test case. I'd prefer we remove it. Not a deal breaker.